app/models/feedback.rb

```diff
@@ -182,6 +183,28 @@ def check_for_dupe_feedback
     throw :abort if duplicate.exists? && !is_invalidated
   end
 
+  def check_for_dupe_feedback_again
+    return if is_invalidated
+    duplicate = if user_id.present?
+                  Feedback.where(user_id: user_id, post_id: post_id)
+                else
+                  Feedback.where(user_name: user_name, post_id: post_id)
+                end.where(is_invalidated: false).where.not(id: id)
+    return unless duplicate.exists?
+    user = if user_id.present?
+             User.find user_id
+           else
+             User.where(user_name: user_name)
+           end
+    duplicate.each do |d|
+      if d.created_at > 1.day.ago
+        d.destroy
```

Comment:
It appears you're destroying the feedback which is > 1 day old and invalidating the feedback which is < 1 day old. That's the opposite of what is desired.

Reply:
irb(main):009:0> DateTime.now > 1.day.ago
=> true

Reply:
doh... brain fart on my part. Thanks.